step/step.go

```diff
@@ -68,24 +71,37 @@ func (d IntermediateFileDownloader) ProcessConfig() (Config, error) {
 		return Config{}, ErrMissingAccessToken
 	}
 
-	finishedStages := input.FinishedStages
+	if input.FinishedStages == "" && input.FinishedWorkflows == "" {
```

Comment:
Maybe guard against having both as well, not just pick one in that case "randomly"